src/test/java/net/snowflake/ingest/streaming/internal/ParquetFlushServiceTest.java

```diff
@@ -0,0 +1,203 @@
+package net.snowflake.ingest.streaming.internal;
+
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import net.snowflake.client.jdbc.internal.org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+public class ParquetFlushServiceTest {
+
+  FlushServiceTest.TestContextFactory<List<List<Object>>> testContextFactory;
+  FlushServiceTest.TestContext<List<List<Object>>> testContext;
+
+  @Before
+  public void setup() {
+    java.security.Security.addProvider(new BouncyCastleProvider());
+    testContextFactory = FlushServiceTest.ParquetTestContext.createFactory();
+    testContext = testContextFactory.create();
+  }
+
+  private SnowflakeStreamingIngestChannelInternal<?> addChannel() {
+    return testContext
+        .channelBuilder("channel1")
+        .setDBName("db1")
+        .setSchemaName("schema1")
+        .setTableName("table1")
+        .setOffsetToken("offset1")
+        .setChannelSequencer(0L)
+        .setRowSequencer(0L)
+        .buildAndAdd();
+  }
+
+  private SnowflakeStreamingIngestChannelInternal<?> addChannel2() {
+    return testContext
+        .channelBuilder("channel2")
+        .setDBName("db1")
+        .setSchemaName("schema1")
+        .setTableName("table1")
+        .setOffsetToken("offset2")
+        .setChannelSequencer(10L)
+        .setRowSequencer(100L)
+        .buildAndAdd();
+  }
+
+  @Test
+  public void testBuildAndUploadMoreDataTypes() throws Exception {
```

Comment:
What is Parquet specific in this test that is not covered by `FlushServiceTest`?
if there is something that is not covered in `FlushServiceTest` but it is not Parquet specific, should it rather be in `FlushServiceTest`?
If it just tests various data types at once then I am not sure it adds much to the coverage as we already have a similar test in IT that is fast enough.

Reply:
It tests various data types at once. You are right, I checked the coverage of both this test and `StreamingIngestIT`. Since this test doesn't add any coverage, I will remove it.

Reply:
btw I just noticed we do not actually check the values returned from the select query in IT
it would be nice to add such a check at some point

Reply:
I added a table count verification `select count(*)` and `select * where ` for the `StreamingIngestIT.testDataTypes` . But the select * fails with `net.snowflake.client.jdbc.SnowflakeSQLException: Time '19:40:55.-226005504' is not recognized`

Reply:
Indeed, the test data is generated incorrectly, actually for all time data types in `getRandomRow`:
```
(14 * 60 * 60 + 26 * 60 + 34) * 1000000000 + 437582643
```
but should be:
```
(14 * 60 * 60 + 26 * 60 + 34) + 0.437582643
```
or the same but a string representation "123134.2343" for `BigDecimals`
maybe a good idea to not make it random and assert expected query result

Reply:
> maybe a good idea to not make it random and assert expected query result

done